express-crud/server.ts

```diff
@@ -0,0 +1,107 @@
+import express from 'express';
+import pg from 'pg';
+import { errorMiddleware } from './lib/index.js';
+import { ClientError } from './lib/client-error.js';
+
+type Grade = {
+  gradeId?: number;
+  name: string;
+  course: string;
+  score: number; // A number between 0 and 100
+};
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false,
+  },
+});
+
+const app = express();
+
+app.use(express.json());
+
+// get the api grades
+app.get('/api/grades', async (req, res, next) => {
+  try {
+    const result = await db.query('SELECT * FROM "grades"');
+    const grades = result.rows;
+    res.status(200).json(grades);
+  } catch (error) {
+    next(new ClientError(500, 'failed to fetch grades'));
+
+  }
+});
+
+//GET grade ID
+app.get('/api/grades/:gradeId', async (req, res, next) => {
+  const { gradeId } = req.params;
```

Comment:
should be a client error here for the gradeId, and that it should be a number and it should throw back a 400 status code